src/stdio.jl

```diff
@@ -52,6 +78,34 @@ function watch_stream(rd::IO, name::AbstractString)
     end
 end
 
+function send_stdio(name)
+    if verbose::Bool && !haskey(task_local_storage(), :IJulia_task)
+        task_local_storage(:IJulia_task, "send $name task")
+        @vprintln("$name task labelled")
+    end
+    send_stream(name)
+end
+
+send_stdout(t::Timer) = send_stdio("stdout")
+send_stderr(t::Timer) = send_stdio("stderr")
+
+function send_stream(name::AbstractString)
+    buf = bufs[name]
+    if buf.size > 0
+        s = takebuf_string(buf)
+        if isvalid(s)
+            send_ipython(publish,
+                     msg_pub(execute_msg, "stream",
+                             @compat Dict("name" => name, "text" => s)))
+        else
+            #if the string is not valid, maybe it's the start of a partially encoded UTF character
+            #so we write it back to the buffer and hope the rest of the character will arrive soon
+            @vprintln("Invalid UTF string written to ", name, ". Bytes: ", convert(Vector{UInt8}, s))
```

Comment:
Oh, duh. Just checking the last char is a much more sane way to do it.
PS I put the debug in because I was worried about people wondering where all their output went in case of an invalid char, so will remove it now.